src/Integrators.jl

```diff
@@ -282,7 +289,7 @@ function step!(sys::System{0}, integrator::ImplicitMidpoint)
         # improved midpoint estimator s̄′.
         @. ŝ = normalize_dipole(s̄, sys.κs)
         set_energy_grad_dipoles!(∇E, ŝ, sys)
-        @. s̄′ = s + 0.5 * Δt * rhs_dipole(ŝ, -∇E)
+        @. s̄′ = s + 0.5 * (Δt * rhs_dipole(ŝ, -∇E, λ) + √Δt * rhs_dipole(ŝ, ξ))
```

Comment:
New call to `rhs_dipole(ŝ, ξ)` might incur nontrivial computational cost. This could be broken into a separate block:

```julia
if kT > 0
    @. s̄′ += √(Δt*2λ*kT) * rhs_dipole(ŝ, ξ))
end
```

And then remove the line `ξ .*= √(2λ*kT)` above.  Note that this scaling of `ξ` as a pure Gaussian is more consistent with the SU(N) case.